test/unit/org/apache/cassandra/tools/AsyncProfilerServiceTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.tools;
+
+import java.io.File;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.After;
+import org.junit.Test;
+
+import org.apache.cassandra.tools.profiler.AsyncProfilerService;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.ASYNC_PROFILER_ENABLED;
+import static org.apache.cassandra.config.CassandraRelevantProperties.ASYNC_PROFILER_ADVANCED_MODE;
+
+public class AsyncProfilerServiceTest
+{
+
+    private static AsyncProfilerService profiler;
+    private static String testOutputFile = "/tmp/test-profile.html";
+
+    @BeforeClass
+    public static void setUpClass()
+    {
+        profiler = new AsyncProfilerService();
+
+        if (!profiler.isAvailable())
+        {
+            fail("AsyncProfilerService could not initialize (native lib not found or invalid).");
+        }
+    }
+
+    @Before
+    public void setUp()
+    {
+        ASYNC_PROFILER_ENABLED.setBoolean(true);
```

Comment:
should we use org.apache.cassandra.distributed.shared.WithProperties?

Reply:
All the tests are needed that flag to be true. I think this setup helps with code duplication?